### ISSUE TITLE
Fetching Card holder's details in case of Checkout Web SDK

### ISSUE THREAD

**Issue by vinaygangaraj** [Adyen/adyen-web-sdk-sample-code#11]
Hi Team,

I am using Web SDK checkout in our application to integrate Cards and Alipay(Local payment methods). I am calling https://checkout-test.adyen.com/services/PaymentSetupAndVerification/v32/setup to get the html and response is binding to the <div> element on our page correctly.

On the rendered html page, if i select the card option and once i enter all the card details and i click on pay button, I need to get the shopper's card details. For this purpose i am using the below hook. But, this is not getting fired when i click on pay button. Please let me know if there is any change is necessary to be made to get the card details? 

chckt.hooks.getDataFromSubmissionProcess = function (checkoutHtmlNode, formData) {

return true;
}

Basically, I need the card holder's name and card details to be saved on our DB. Is there any way so that i can get these details either from Adyen server or from using sdks?

Thanks,
Vinay

**Comment by Trigalti** [comment]
Hey Vinay, in checkout web SDK this is not an option.
To be able to sture card details in your database, you have to be PCI compliant.

In the case of being able to store cardHolder name and card details, you would have to integrate with our API, become PCI compliant and start sending in data through back-end:
https://docs.adyen.com/developers/checkout/api-integration

Kind regards,

Simon

**Comment by vinaygangaraj** [comment]
Hi,

Thanks for your reply. 

Since i am using checkout Web SDK, for card option when shopper enters all the data and when he clicks on pay button the data is encrypted and sent to Adyen. So, Is there anyway where we can decrypt the card details and save on our DB before it goes to adyen ? 

Basically i need the last 4 digits of the card, expiry date, card holder's name and card brand. And, I need this without being PCI compliant Level1 or Level2. Because, if i opt for Checkout API Integration we need to  be PCI compliant Level1 or Level2. 

Or does Adyen send card details (last 4 digits of the card, expiry date, card holder's name and card brand) in the response ?

Thanks,
Vinay

**Comment by Trigalti** [comment]
Hey Vinay,

Unfortunately there is no way to decrypt this data (unless you send this in unencrypted).
There are two ways of receiving the last 4 digits, expiry date etc;

If you send in a shopperReference when making the setup call, a "recurring/oneclick" contract will be created.
On an additional setup call with the same shopperReference, you will see the recurring details showing up in the interface.

Another way is to go API and use the getPaymentMethods endpoint:
https://docs.adyen.com/api-explorer/#/PaymentSetupAndVerificationService/32/paymentMethods

When sending in a shopperReference, any stored details will be returned.

Kind regards,

Simon

**Comment by vinaygangaraj** [comment]
Awesome... Thanks a lot for your help :)

I have one last question to be asked, 

In this screenshot as you can see that Card Holder Name field is missing https://www.screencast.com/t/KnoWSXO4Rrt

The above html is generated by Adyen, also in the set up response i could see that {"key":"holderName","**optional":"true",**"type":"text"}. Since it is optional i think we are not able to see the cardholder name in the UI. Could you please help me out on how to render Card holder name in the UI?

{"group":{"name":"Credit Card","paymentMethodData":"Cf62f1e3!YnJhbmRDb2RlPXNjaGVtZQ==","type":"card"},"inputDetails":[{"key":"encryptedCardNumber","type":"cardToken"},{"key":"encryptedSecurityCode","type":"cardToken"},{"key":"encryptedExpiryMonth","type":"cardToken"},{"key":"encryptedExpiryYear","type":"cardToken"},{"key":"holderName","optional":"true","type":"text"},{"key":"storeDetails","optional":"true","type":"boolean"}],"name":"Diners Club","paymentMethodData":"Cf62f1e3!YnJhbmRDb2RlPWRpbmVycw==","type":"diners"}

Please note, just now i found that we can get the card details in the response by enabling the configuration settings under API URLs and Response tab (Under Settings). :)

Thanks,
Vinay
